includes/admin/class-wc-rest-stripe-payment-gateway-controller.php

```diff
@@ -1,175 +0,0 @@
-				'permission_callback' => [ $this, 'check_permission' ],
-			]
-		);
-	}
-
-	/**
-	 * Retrieve payment gateway settings.
-	 *
-	 * @param WP_REST_Request $request
-	 * @return WP_REST_Response
-	 */
-	public function get_payment_gateway_settings( $request = null ) {
-		try {
-			$id = $request->get_param( 'payment_gateway_id' );
-			$this->instantiate_gateway( $id );
-			$settings = [
-				'is_' . $id . '_enabled' => $this->gateway->is_enabled(),
-				$id . '_name'            => $this->gateway->get_option( 'title' ),
-				$id . '_description'     => $this->gateway->get_option( 'description' ),
-			];
-			$settings = $this->gateway->get_unique_settings( $settings );
-			return new WP_REST_Response( $settings );
-		} catch ( Exception $exception ) {
-			return new WP_REST_Response( [ 'result' => 'bad_request' ], 400 );
-		}
-	}
-
-	/**
-	 * Update payment gateway settings.
-	 *
-	 * @param WP_REST_Request $request Request object.
-	 */
-	public function update_payment_gateway_settings( WP_REST_Request $request ) {
-		try {
-			$id = $request->get_param( 'payment_gateway_id' );
-			$this->instantiate_gateway( $id );
-			$this->update_is_gateway_enabled( $request );
-			$this->update_gateway_name( $request );
-			$this->update_gateway_description( $request );
-			if ( method_exists( $this->gateway, 'update_unique_settings' ) ) {
-				$this->gateway->update_unique_settings( $request );
-			}
-			return new WP_REST_Response( [], 200 );
-		} catch ( Exception $exception ) {
-			return new WP_REST_Response( [ 'result' => 'bad_request' ], 400 );
-		}
-	}
-
-	/**
-	 * Updates payment gateway enabled status.
-	 *
-	 * @param WP_REST_Request $request Request object.
-	 */
-	private function update_is_gateway_enabled( WP_REST_Request $request ) {
-		$field_name = 'is_' . $this->gateway->id . '_enabled';
-		$is_enabled = $request->get_param( $field_name );
-
-		if ( null === $is_enabled || ! is_bool( $is_enabled ) ) {
-			return;
-		}
-
-		if ( $is_enabled ) {
-			$this->gateway->enable();
-		} else {
-			$this->gateway->disable();
-		}
-	}
-
-	/**
-	 * Updates payment gateway title.
-	 *
-	 * @param WP_REST_Request $request Request object.
-	 */
-	private function update_gateway_name( WP_REST_Request $request ) {
-		$field_name = $this->gateway->id . '_name';
-		$name       = $request->get_param( $field_name );
-
-		if ( null === $name ) {
-			return;
-		}
-
-		$value = sanitize_text_field( $name );
-		$this->gateway->update_option( 'title', $value );
-	}
-
-	/**
-	 * Updates payment gateway description.
-	 *
-	 * @param WP_REST_Request $request Request object.
-	 */
-	private function update_gateway_description( WP_REST_Request $request ) {
-		$field_name  = $this->gateway->id . '_description';
-		$description = $request->get_param( $field_name );
-
-		if ( null === $description ) {
-			return;
-		}
-
-		$value = sanitize_text_field( $description );
-		$this->gateway->update_option( 'description', $value );
-	}
-}
```

Comment:
As all the methods are moved inside Stripe settings page, this separate page for the individual methods is not needed anymore. 

![separate](https://github.com/woocommerce/woocommerce-gateway-stripe/assets/33387139/f35de1f9-b2d6-4f33-9e6e-0cb58816e42b)


So deleting the controller and relevant tests. Also the tests started to fail as we are not registering any payment method for the payment settings page.